doc/mlsys_paper/paper.tex

```diff
@@ -301,48 +305,50 @@ \section{Types of objective functions}
 \vspace*{-0.3em}
 
 In this framework, if a user wants to optimize a differentiable objective
-function, they need only provide implementations of $f(x)$ and $f'(x)$, and
+function, they only need to provide implementations of $f(x)$ and $f'(x)$, and
 then they can use any of the gradient-based optimizers that {\tt ensmallen}
-provides.  Table~\ref{tab:functionality} compares
-the classes of objective functions that {\tt ensmallen} can handle with other
-popular frameworks and libraries.
+provides.  Table~\ref{tab:functionality} contrasts 
+the classes of objective functions that can be handled by {\tt ensmallen}
+and other popular frameworks and libraries.
 
-Of course, not every optimization algorithm provided by {\tt ensmallen} can be
+Not every optimization algorithm provided by {\tt ensmallen} can be
 used by every class of objective function; for instance, a gradient-based
 optimizer such as L-BFGS cannot operate on a non-differentiable objective
 function.
-
-Therefore, the best the library can attain is to maximize the flexibility
+As such, the best the library can attain is to maximize the flexibility
 available, so that a user can easily implement a function $f(x)$ and have it
 work with as many optimizers as possible.
-To do this, {\tt ensmallen} makes heavy use of C++ template metaprogramming.
-When implementing an objective function to be
-optimized, a user can implement only a few methods and we can use
-metaprogramming to check that the given functions match the requirements of the
+
+To accomplish the flexibility, {\tt ensmallen} makes heavy use of
+C++ template metaprogramming.
+When implementing an objective function to be optimized,
+a user can implement only a few methods; metaprogramming is then automatically
```

Comment:
Do you mind if I change "can implement" to "may implement" here?  This wording could be misread to imply that a user is only allowed to implement a few methods, which sounds restrictive.

Reply:
ok with me.  how about:
"a user may only need to implement a few methods"
this sounds more positive in the sense that it implies that not a lot of work is needed